pkg/apis/kappctrl/v1alpha1/types_template.go

```diff
@@ -68,16 +73,35 @@ type AppTemplateValuesSourceRef struct {
 	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
 }
 
+// +k8s:openapi-gen=true
+type AppTemplateValuesDownwardAPI struct {
+	Items []AppTemplateValuesDownwardAPIItem `json:"items,omitempty" protobuf:"bytes,1,opt,name=items"`
+}
+
 // +k8s:openapi-gen=true
 type AppTemplateValuesDownwardAPIItem struct {
 	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
 	// Required: Selects a field of the app: only annotations, labels, uid, name and namespace are supported.
 	FieldPath string `json:"fieldPath,omitempty" protobuf:"bytes,2,opt,name=fieldPath"`
+	// Optional: Get running Kubernetes version from cluster, defaults (empty) to retrieving the version from the cluster.
+	// Can be manually supplied instead.
+	KubernetesVersion *Version `json:"kubernetesVersion,omitempty" protobuf:"bytes,3,opt,name=kubernetesVersion"`
+	// Optional: Get running KappController version, defaults (empty) to retrieving the current running version..
+	// Can be manually supplied instead.
+	KappControllerVersion *Version `json:"kappControllerVersion,omitempty" protobuf:"bytes,4,opt,name=kappControllerVersion"`
+	// Optional: Get running KubernetesAPIs from cluster, defaults (empty) to retrieving the APIs from the cluster.
+	// Can be manually supplied instead, e.g ["group/version", "group2/version2"]
+	KubernetesAPIs *GroupVersion `json:"kubernetesAPIs,omitempty" protobuf:"bytes,5,opt,name=kubernetesAPIs"`
 }
 
 // +k8s:openapi-gen=true
-type AppTemplateValuesDownwardAPI struct {
-	Items []AppTemplateValuesDownwardAPIItem `json:"items,omitempty" protobuf:"bytes,1,opt,name=items"`
+type Version struct {
```

Comment:
i would recommend renaming this to KubernetesVersion and KubernetesGroupVersions

Reply:
I renamed GroupVersions to KubernetesGroupVersions but I left the other as Version as it made more sense to me to keep them together / and it's a key underneath `KubernetesVersion` so it felt fine to leave as is